test/solc/CommandLineInterface.cpp

```diff
@@ -1412,6 +1412,130 @@ BOOST_AUTO_TEST_CASE(cli_include_paths_ambiguous_import)
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--ethdebug", "--via-ir", "--import-ast", tempDir.path().string() + "/input.json"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ir", "--import-ast", tempDir.path().string() + "/input.json"}, "");
+	BOOST_REQUIRE(!result.success);
+}
+
+BOOST_AUTO_TEST_CASE(cli_ethdebug_debug_info_ethdebug)
+{
+	TemporaryDirectory tempDir(TEST_CASE_NAME);
+	createFilesWithParentDirs({tempDir.path() / "input.sol"},  "pragma solidity >=0.0; contract C { function f() public pure {} }");
+	createFilesWithParentDirs({tempDir.path() / "input.yul"}, "{}");
+	// --debug-info ethdebug only works with --ir, --ir-optimized and --ethdebug (--ethdebug always need --via-ir)
+	OptionsReaderAndMessages result = runCLI({"solc", "--debug-info", "ethdebug", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	// debug-info selection "ethdebug" only works, if "ir" and "irOptimized" was selected.
+	result = runCLI({"solc", "--debug-info", "ethdebug", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ir", tempDir.path().string() + "/input.sol"}, "");
+	// if --debug-info ethdebug was set, we will output a comment on top of the ir output, indicating that ethdebug was enabled,
+	// so we use this to check whether --debug-info ethdebug was implicitly enabled.
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") != std::string::npos);
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ir-optimized", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") != std::string::npos);
+	// --ethdebug only works, if --via-ir was defined
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ethdebug", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ethdebug", "--via-ir", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("Debug Data (ethdebug):") != std::string::npos);
+	// --debug-info ethdebug always works with --strict-assembly
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--strict-assembly", tempDir.path().string() + "/input.yul"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") != std::string::npos);
+	// --debug-info ethdebug is excluded from --debug-info all
+	result = runCLI({"solc", "--debug-info", "all", "--strict-assembly", tempDir.path().string() + "/input.yul"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") == std::string::npos);
+}
+
+BOOST_AUTO_TEST_CASE(cli_ethdebug_ethdebug_output)
```

Comment:
So, given that you're covering the existence of `/// ethdebug: enabled` in generated output for different CLI combinations, it's definitely better that they're all here instead of in a million cmdlineTests with massive outputs - but it would none the less be good to have at least one cmdlineTest that should the full generated output (a smoke test if you will).

Reply:
Ah thats true! Will add that soon